test/output.go

```diff
@@ -686,3 +699,230 @@ func (e *customErrorJSON) Error() string {
+{"caller":"test/output.go:<WITH-VALUES-2>","msg":"odd number of arguments passed as key-value pairs for logging","ignored key":"anotherKeyWithoutValue"}
+{"caller":"test/output.go:<LINE>","msg":"odd WithValues","v":0}
+{"caller":"test/output.go:<LINE>","msg":"odd WithValues","v":0}
+{"caller":"test/output.go:<LINE>","msg":"odd WithValues","v":0}
+`,
+
+		`I output.go:<LINE>] "odd arguments" akey="avalue" akey2="(MISSING)"
+`: `{"caller":"test/output.go:<LINE>","msg":"odd number of arguments passed as key-value pairs for logging","ignored key":"akey2"}
+{"caller":"test/output.go:<LINE>","msg":"odd arguments","v":0,"akey":"avalue"}
+`,
+
+		`I output.go:<LINE>] "both odd" basekey1="basevar1" basekey2="(MISSING)" akey="avalue" akey2="(MISSING)"
+`: `{"caller":"test/output.go:<WITH-VALUES>","msg":"odd number of arguments passed as key-value pairs for logging","ignored key":"basekey2"}
+{"caller":"test/output.go:<LINE>","msg":"odd number of arguments passed as key-value pairs for logging","basekey1":"basevar1","ignored key":"akey2"}
+{"caller":"test/output.go:<LINE>","msg":"both odd","basekey1":"basevar1","v":0,"akey":"avalue"}
+`,
+
+		// klog.Info
+		`I output.go:<LINE>] "helloworld\n"
+`: `{"caller":"test/output.go:<LINE>","msg":"helloworld\n","v":0}
+`,
+
+		// klog.Infoln
+		`I output.go:<LINE>] "hello world\n"
+`: `{"caller":"test/output.go:<LINE>","msg":"hello world\n","v":0}
+`,
+
+		// klog.Error
+		`E output.go:<LINE>] "helloworld\n"
+`: `{"caller":"test/output.go:<LINE>","msg":"helloworld\n"}
+`,
+
+		// klog.Errorln
+		`E output.go:<LINE>] "hello world\n"
+`: `{"caller":"test/output.go:<LINE>","msg":"hello world\n"}
+`,
+
+		// klog.ErrorS
+		`E output.go:<LINE>] "world" err="hello"
+`: `{"caller":"test/output.go:<LINE>","msg":"world","err":"hello"}
+`,
+
+		// klog.InfoS
+		`I output.go:<LINE>] "hello" what="world"
+`: `{"caller":"test/output.go:<LINE>","msg":"hello","v":0,"what":"world"}
+`,
+
+		// klog.V(1).Info
+		`I output.go:<LINE>] "hellooneworld\n"
+`: `{"caller":"test/output.go:<LINE>","msg":"hellooneworld\n","v":1}
+`,
+
+		// klog.V(1).Infoln
+		`I output.go:<LINE>] "hello one world\n"
+`: `{"caller":"test/output.go:<LINE>","msg":"hello one world\n","v":1}
+`,
+
+		// klog.V(1).ErrorS
+		`E output.go:<LINE>] "one world" err="hello"
+`: `{"caller":"test/output.go:<LINE>","msg":"one world","err":"hello"}
+`,
+
+		// klog.V(1).InfoS
+		`I output.go:<LINE>] "hello" what="one world"
+`: `{"caller":"test/output.go:<LINE>","msg":"hello","v":1,"what":"one world"}
+`,
+	}
+}
+
+// ZaprOutputMappingIndirect provides a mapping from klog output to the
+// corresponding zapr output when zapr is called indirectly through
+// klog.
+//
+// This is different from ZaprOutputMappingDirect because:
+// - WithName gets added to the message by Output.
+// - WithValues are part of the normal key/value parameters,
+//   which puts them after "v".
+// - There are no separate warnings about WithValue parameters.
```

Comment:
... when using zap directly.

I need to make that comment clearer. It might also make sense to add that to zapr, for consistency -> https://github.com/go-logr/zapr/issues/51

Reply:
It's the other way around: [zapr emits a warning](https://github.com/go-logr/zapr/blob/517e434e7dd06b6a09644da072c246db038a8c62/zapr.go#L145), [klog doesn't](https://github.com/kubernetes/klog/blob/b7a44e3874816d2b62aee5947c4241e316db14b7/internal/serialize/keyvalues.go#L61-L65).

Should we add that to klog?